Tools/autotest/common.py

```diff
@@ -594,7 +594,7 @@ def wait_mode(self, mode, timeout=None):
                 hastimeout = self.get_sim_time() > tstart + timeout
             self.mav.wait_heartbeat()
         self.progress("Got mode %s" % mode)
-        return self.mav.flightmode
+        return self.mav.flightmode.upper() == mode.upper()
```

Comment:
This would be better if it checked `mode` was actually in the known modes (and removing the `.upper()` canonicalisation).  That catches spelling errors as well as case problems.